test/framework/test_composite.py

```diff
@@ -365,6 +367,35 @@ def test_composite_tags(self):
         self.assertEqual(sorted(data1.tags), ["c", "d"])
         self.assertEqual(sorted(data2.tags), ["c", "d"])
 
+    def test_composite_figures(self):
+        """
+        Test adding figures from composite experiments
+        """
+        exp1 = FakeExperiment([0, 2])
+        exp2 = FakeExperiment([1, 3])
+        exp1.analysis.set_options(add_figures=True)
+        exp2.analysis.set_options(add_figures=True)
+        par_exp = BatchExperiment([exp1, exp2], flatten_results=True)
+        expdata = par_exp.run(FakeBackend())
+        expdata.service = IBMExperimentService(local=True, local_save=False)
+        expdata.auto_save = True
+        expdata.block_for_results()
+        self.assertExperimentDone(expdata)
```

Comment:
This test passes on the current main branch too, is it testing figures sufficiently?

Reply:
The new test is still passing on the main branch for me.

Reply:
Thanks, I updated the test to ensure the scenario where it attempts to do `add_figures` on the figures already added by the sub-experiments (and hence, wrapped in the `FigureData` container).

When testing this on the main branch, note that `fake_experiment.py` must be updated as well, otherwise the test won't generate figures.